foursight_core/scripts/local_check_execution.py

```diff
@@ -0,0 +1,203 @@
+            for arg_name in check_or_action_info.kwargs:
+                arg_default = check_or_action_info.kwargs[arg_name]
+                value = input(f"Enter {kind} argument => "
+                              f"{arg_name} [default: {arg_default}]: ")
+                args[arg_name] = value
+        if verbose:
+            if args:
+                print(f"{kind.title()} arguments for {name}:")
+                for arg in args:
+                    print(f"=> {arg}: {args[arg]}")
+    return args
+
+
+def find_check_or_action(app_utils, check_or_action) -> Tuple[Optional[object], Optional[object]]:
+    """
+    Given a check or action name return the "info" object for one or the other depending
+    on whether a check or action name was specified. These names are meant to be fully
+    qualfied, e.g. "access_key_expiration_detection/access_key_status". First tries
+    to match check and then action exactly, and then if not found then tries an
+    approximate match (if given name is part of the full check/action name),
+    first trying the check name and then the action name.
+    """
+    check_info = app_utils.check_handler.get_check_info(check_or_action)
+    action_info = None
+    if not check_info:
+        # Maybe they specified an action instead.
+        action_info = app_utils.check_handler.get_action_info(check_or_action)
+        if not action_info:
+            checks_info = app_utils.check_handler.get_checks_info(check_or_action)
+            if checks_info and len(checks_info) == 1:
+                message = f"Did you mean to specify this check: {checks_info[0].qualified_name}?"
+                if confirm_interactively(message):
+                    check_info = checks_info[0]
+            if not check_info:
+                actions_info = app_utils.check_handler.get_actions_info(check_or_action)
+                if actions_info and len(actions_info) == 1:
+                    message = f"Did you mean to specify this action: {actions_info[0].qualified_name}?"
+                    if confirm_interactively(message):
+                        action_info = actions_info[0]
+        if not check_info and not action_info:
+            exit_with_no_action(f"No check or action found: {check_or_action}")
+    return check_info, action_info
+
+
+def guess_env() -> Optional[str]:
+    aws_test_dir_name = ".aws_test"
+    aws_test_dir_prefix = f"{aws_test_dir_name}."
+    aws_test_dir = os.path.expanduser(f"~/{aws_test_dir_name}")
+    if os.path.islink(aws_test_dir):
+        aws_test_target_dir = os.readlink(aws_test_dir)
+        if os.path.exists(aws_test_target_dir):
+            aws_test_target_dir = os.path.basename(aws_test_target_dir)
+            if aws_test_target_dir.startswith(aws_test_dir_prefix):
+                aws_credentials_name = aws_test_target_dir[len(aws_test_dir_prefix):]
+                return aws_credentials_name
+
+
+def confirm_interactively(message: str, exit_if_no: bool = False) -> bool:
```

Comment:
You have code very similar to this in 4dn-cloud-infra you could also bring into utils ie: arg_utils ?